lib/vmpooler/pool_manager.rb

```diff
@@ -40,25 +39,38 @@ def _check_pending_vm(vm, pool, timeout, vsphere)
 
       if host
         begin
-          open_socket vm
+          socket = open_socket vm
+          fail_pending_vm(vm, pool, timeout) if ! socket
           move_pending_vm_to_ready(vm, pool, host)
         rescue
           fail_pending_vm(vm, pool, timeout)
         end
       else
-        fail_pending_vm(vm, pool, timeout)
+        fail_pending_vm(vm, pool, timeout, false)
       end
     end
 
-    def fail_pending_vm(vm, pool, timeout)
-      clone_stamp = $redis.hget('vmpooler__vm__' + vm, 'clone')
-
-      if (clone_stamp) &&
-          (((Time.now - Time.parse(clone_stamp)) / 60) > timeout)
-
-        $redis.smove('vmpooler__pending__' + pool, 'vmpooler__completed__' + pool, vm)
+    def remove_nonexistent_vm(vm, pool)
+      $redis.srem("vmpooler__pending__#{pool}", vm)
+      $logger.log('d', "[!] [#{pool}] '#{vm}' no longer exists. Removing from pending.")
+    end
 
-        $logger.log('d', "[!] [#{pool}] '#{vm}' marked as 'failed' after #{timeout} minutes")
+    def fail_pending_vm(vm, pool, timeout, exists=true)
+      begin
```

Comment:
That's handy. I have updated the fail_pending_vm method accordingly. Let me know if it doesn't look right.